# HG changeset patch
# User Jaroslav Hajek <highegg@gmail.com>
# Date 1249896959 -7200
#      Mon Aug 10 11:35:59 2009 +0200
# Node ID 1b290ce305fbfdef954fc158c09d346276aafc84
# Parent  c5330ef7aecde74f4300fdde64f8259fa6501ea3
add omitted check in flipdim

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2009-08-07  Jaroslav Hajek  <highegg@gmail.com>
+
+	* general/flipdim.m: Fix omitted check.
+
 2009-08-07  Jaroslav Hajek  <highegg@gmail.com>
 
 	* general/flipdim.m: Simplify.
 
 2009-08-07  Olli Saarela  <olli.saarela@gmail.com>
 
 	* plot/__gnuplot_ginput__.m: If read fails to return data, sleep
 	before trying again.
diff --git a/scripts/general/flipdim.m b/scripts/general/flipdim.m
--- a/scripts/general/flipdim.m
+++ b/scripts/general/flipdim.m
@@ -43,17 +43,17 @@ function y = flipdim (x, dim)
   nd = ndims (x);
   if (nargin == 1)
     ## Find the first non-singleton dimension.
     dim = find (size (x) != 1, 1);
     if (isempty (dim))
       dim = 1;
     endif
   else
-    if (! isindex (dim, nd))
+    if (! (isscalar (dim) && isindex (dim, nd)))
       error ("flipdim: dim must be an integer and valid dimension");
     endif
   endif
 
   idx(1:nd) = {':'};
   idx{dim} = size (x, dim):-1:1;
   y = x(idx{:});
 
